src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -380,12 +380,56 @@ private static void AssertFloatingPointBehavior<T>(T netfxExpectedValue, T netco
         public static void LongInputString(int length)
         {
             // Verify boundary conditions in Deserialize() that inspect the size to determine allocation strategy.
-            string repeated = new string('x', length - 2);
-            string json = $"\"{repeated}\"";
-            Assert.Equal(length, json.Length);
+            DeserializeLongJsonString(length);
+        }
+
+        private const int MaxInt = int.MaxValue / MaxExpansionFactorWhileTranscoding;
+        private const int MaximumPossibleStringLength = int.MaxValue / 2 - 32;
+
+        // NOTE: VeryLongInputString test is constrained to run on Windows and MacOSX because it causes
+        //       problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
+        //       succeed even if there is not enough memory but then the test may get killed by the OOM killer at the
+        //       time the memory is accessed which triggers the full memory allocation.
+        [ConditionalTheory(nameof(IsX64))]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [InlineData(MaxInt - 3)]
+        [InlineData(MaxInt - 2)]
+        [InlineData(MaxInt - 1)]
+        [InlineData(MaxInt)]
+        [InlineData(MaxInt + 1)]
+        [InlineData(MaxInt + 2)]
+        [InlineData(MaxInt + 3)]
+        [InlineData(MaximumPossibleStringLength - 3)]
+        [InlineData(MaximumPossibleStringLength - 2)]
+        [InlineData(MaximumPossibleStringLength - 1)]
+        [InlineData(MaximumPossibleStringLength)]
```

Comment:
Why do we need all of these variations?  These 11 additions cause the test suite on my machine to run for an additional 30 seconds.  Are all of these contributing meaningfully?  Why can't this just be a single input of the longest length here?

Reply:
What about all the variations on MaxInt?

Reply:
I think those are useful to keep since they will help avoid off-by-one type bugs in the buffer creation. If it is too costly (even as outerloop), we could remove the +2, 3 and -2, 3 cases.

Reply:
What kind of off-by-one are you concerned about (or off-by-two or off-by-three)?  Seems like the test is just trying to deserialize something really big, so we can have a single test that validates the longest thing we hope to work works... no?

Reply:
Code like this:
https://github.com/dotnet/corefx/blob/17d22ab65efa048b2971d2d1c93cb8bfa54635b4/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.String.cs#L75-L80

Reply:
So then don't we need a total of two tests, one known to hit one path and one known to hit the other?  What will additional cases help with?

Reply:
Depends on what the intent of these tests is. If we are testing directly against the current implementation, then you are right, a smaller set can cover all the branches. However, if we assume the implementation could change and there could be bugs introduced, then having test cases that are +/- around some of the key "constants" being used is generally, good, no?

To put differently, I don't see much harm in being extra defensive with some test cases (within reason).

Up to you. I don't feel too strongly about this case (I'd be fine with this PR merged as is, and would also be OK with a reduced set).

Reply:
> then having test cases that are +/- around some of the key "constants" being used is generally, good, no?

To know that e.g. +2 and -3 are useful, you need to know the implementation. If the implementation could change, why is it likely that those values are still meaningful/useful? If I decide to change the threshold from e.g. 512 to 1024, these offsets are just as wrong as the core value, and I'm much more likely to make such a threshold change than to change e.g. 512 to 514.

I'm all for having tests that validate distinct cases, e.g. when writing into a span, validating the <, ==, and > cases for length. But that doesn't appear to be the case here: we're just trying to exercise two code paths.

If I'm misunderstanding, that's fine, I'll leave it up to you. I just don't like when we have a lot of duplicative test cases that spend time and money executing stuff that doesn't add additional value.

Reply:
That's fair. Given we already have tests around the `Threshold`, just having `MaxInt` makes sense.

@eugenesmlv, would you mind removing the extra cases? Ty.
```C#
	    [InlineData(MaxInt - 3)]
        [InlineData(MaxInt - 2)]
        [InlineData(MaxInt - 1)]
        [InlineData(MaxInt + 1)]
        [InlineData(MaxInt + 2)]
        [InlineData(MaxInt + 3)]
```

Reply:
Sure, and what about tests around the `Threshold`? Should we keep all of these variations?
```csharp
[InlineData(Threshold - 3)]
[InlineData(Threshold - 2)]
[InlineData(Threshold - 1)]
[InlineData(Threshold)]
[InlineData(Threshold + 1)]
[InlineData(Threshold + 2)]
[InlineData(Threshold + 3)]
```

Reply:
I'd leave the existing test cases around `Threshold` as is.

Reply:
Thanks.